setters/parser.go

```diff
@@ -0,0 +1,107 @@
+package setters
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+type (
```

Comment:
could you keep all the type standalone, i mean `type xxx struct {}` rather than multiple type in one `type ()` block